waiter/src/waiter/marathon.clj

```diff
@@ -181,24 +183,24 @@
   "Returns the descriptor to be used by Marathon to create new apps."
   [home-path-prefix service-id->password-fn {:keys [service-id service-description]}]
   (let [health-check-url (sd/service-description->health-check-url service-description)
-        {:strs [grace-period-secs run-as-user]} service-description
+        {:strs [cmd cpus disk grace-period-secs mem ports restart-backoff-factor run-as-user]} service-description
         home-path (str home-path-prefix run-as-user)]
     {:id service-id
      :env (scheduler/environment service-id service-description service-id->password-fn home-path)
      :user run-as-user
-     :cmd (get service-description "cmd")
-     :disk (get service-description "disk")
-     :mem (get service-description "mem")
-     :ports (get service-description "ports" [0])
```

Comment:
I understand what your new code does, but I'm not sure I understand how the old code worked. What if I provided a number like 4 for "ports"?